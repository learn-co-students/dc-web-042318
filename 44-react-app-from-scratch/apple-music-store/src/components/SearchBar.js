import React from 'react';

const SearchBar = (props) => {
  return(
    <input style={{width:'60%'}} placeholder={"Search"} onChange={(e) => {props.searchHandler(e.target.value)}}/>
  );
}

export default SearchBar;
