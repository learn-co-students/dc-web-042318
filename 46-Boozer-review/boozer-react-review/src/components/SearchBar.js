import React from "react";
import { Input } from "semantic-ui-react";

const SearchBar = ({ value, handleInputChange }) => {
  return (
    <Input value={value} onChange={e => handleInputChange(e.target.value)} />
  );
};

export default SearchBar;
