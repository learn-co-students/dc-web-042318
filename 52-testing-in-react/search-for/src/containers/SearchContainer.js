import React from 'react'
import SearchBar from '../components/SearchBar'

class SearchContainer extends React.Component {
	state = {
		query: ''
	}
	render() {
		return (
			<div className="search-container">
				<SearchBar query={this.state.query} />
			</div>
		)
	}
}
export default SearchContainer
