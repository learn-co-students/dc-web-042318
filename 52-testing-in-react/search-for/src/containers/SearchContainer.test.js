import React from 'react'
import SearchContainer from './SearchContainer'
import { shallow } from 'enzyme'
import SearchBar from '../components/SearchBar'

describe('shallow SearchContainer', () => {
	// it('renders succesfully', () => {
	// 	let wrapper = shallow(<SearchContainer />)
	// 	expect(wrapper.find('div').length).toEqual(1)
	// })
	// it('render the SearchBar component', () => {
	// 	let wrapper = shallow(<SearchContainer />)
	// 	expect(wrapper.containsMatchingElement(<SearchBar />)).toEqual(true)
	// })
	it('renders', () => {
		let wrapper = shallow(<SearchContainer />)
		expect(wrapper).toMatchSnapshot()
	})
})
