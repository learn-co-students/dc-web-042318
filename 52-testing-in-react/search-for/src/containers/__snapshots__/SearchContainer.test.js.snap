// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow SearchContainer renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <SearchBar
        query=""
/>,
      "className": "search-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "query": "",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <SearchBar
          query=""
/>,
        "className": "search-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "query": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
